File: plugins/vehicle_plugins/src/truck.cpp
#include "vehicle_plugins/truck.hpp"
#include "pluginlib/class_list_macros.hpp"

namespace vehicle_plugins
{

void Truck::move()
{
    RCLCPP_INFO(rclcpp::get_logger("Truck"), "Truck is moving heavily ! 🚛");
}

void Truck::stop()
{
    RCLCPP_INFO(rclcpp::get_logger("Truck"), "Truck stopped !");
}

std::string Truck::getInfo()
{
    return "Truck - 18 wheels - heavy vehicle";
}

}  // namespace vehicle_plugins

PLUGINLIB_EXPORT_CLASS(vehicle_plugins::Truck, vehicle_base::VehicleBase)

File: plugins/vehicle_plugins_py/vehicle_plugins_py/truck.py
from vehicle_base_py.vehicle_base import VehicleBase

class Truck(VehicleBase):
    def move(self) -> None:
        print("Truck is moving heavily ! 🚛")

    def stop(self) -> None:
        print("Truck stopped !")

    def get_info(self) -> str:
        return "Truck - 18 wheels - heavy vehicle"